website/server/extractDocs.js

```diff
@@ -206,30 +208,140 @@ function componentsToMarkdown(type, json, filepath, idx, styles) {
 
 let componentCount;
 
+function getTypedef(filepath, fileContents, json) {
+  let typedefDocgen;
+  try {
+    typedefDocgen = docgen.parse(
+      fileContents,
+      docgenHelpers.findExportedType,
+      [docgenHelpers.typedefHandler]
+    ).map((type) => type.typedef);
+  } catch (e) {
+    // Ignore errors due to missing exported type definitions
+    if (e.message.indexOf(docgen.ERROR_MISSING_DEFINITION) !== -1) {
+      console.error('Cannot parse file', filepath, e);
+    }
+  }
+  if (!json) {
+    return typedefDocgen;
+  }
+  let typedef = typedefDocgen;
+  if (json.typedef && json.typedef.length !== 0) {
+    json.typedef.forEach(def => {
+      const typedefMatch = typedefDocgen.find(t => t.name === def.name);
+      if (typedefMatch) {
+        typedef.name = merge(typedefMatch, def);
+      } else {
+        typedef.push(def);
+      }
+    });
+  }
+  return typedef;
+}
+
+function merge(obj1, obj2) {
+  if (obj1 == null && obj2 == null) {
+    return null;
+  }
+  const merged = obj1;
+  for (const prop in obj2) {
+    if (obj2[prop] != null) {
+      merged[prop] = obj2[prop];
+    }
+  }
+  return merged;
+}
+
 function renderComponent(filepath) {
+  const fileContent = fs.readFileSync(filepath);
   const json = docgen.parse(
-    fs.readFileSync(filepath),
+    fileContent,
     docgenHelpers.findExportedOrFirst,
     docgen.defaultHandlers.concat([
       docgenHelpers.stylePropTypeHandler,
       docgenHelpers.deprecatedPropTypeHandler,
+      docgenHelpers.jsDocFormatHandler,
     ])
   );
+  json.typedef = getTypedef(filepath, fileContent);
 
   return componentsToMarkdown('component', json, filepath, componentCount++, styleDocs);
 }
 
-function renderAPI(type) {
-  return function(filepath) {
-    let json;
-    try {
-      json = jsDocs(fs.readFileSync(filepath).toString());
-    } catch (e) {
-      console.error('Cannot parse file', filepath, e);
-      json = {};
-    }
-    return componentsToMarkdown(type, json, filepath, componentCount++);
+function isJsDocFormat(fileContent) {
+  const reComment = /\/\*\*[\s\S]+?\*\//g;
+  const comments = fileContent.match(reComment);
+  if (!comments) {
+    return false;
+  }
+  return !!comments[0].match(/\s*\*\s+@jsdoc/);
```

Comment:
Yay! I like the attribute method to figure out whether a docs is in jsdoc format or now.